src/interface/src/app/plan/create-scenarios/create-scenarios.component.ts

```diff
@@ -170,12 +171,20 @@ export class CreateScenariosComponent implements OnInit {
 
   loadConfig(): void {
     this.planStateService.getScenario(this.scenarioId!).subscribe({
-      next: (scenario) => {
+      next: (scenario: Scenario) => {
         // if we have the same state do nothing.
         if (this.scenarioState === scenario.scenario_result?.status) {
           return;
         }
 
+        // Updating breadcrumbs
+        this.scenarioName = scenario?.name || null;
```

Comment:
here you already know that you have scenario, so I dont think you need to `scenario?.name || null;` just do scenario.name ?

Reply:
If found the root cause of why my editor was highlighting all those `|| null `  as required.
The reason was that the property ID was optional in Scenario interface which means it could be ` string || null || undefined` I removed the optional from the Scenario interface and I think now is more clear.
Before:
![image](https://github.com/user-attachments/assets/9dcd3591-8981-435f-ae55-be554d30f100)

Now:
![image](https://github.com/user-attachments/assets/d3fb7fd0-4c1e-4431-819a-7cd264595749)

I have tested it and there is no impact.